classes/manage_content_types.php

```diff
@@ -350,6 +358,19 @@ public function delete(int $id): void {
         }
     }
 
+    /**
+     * Set the visibility of the CMS type.
+     *
+     * @param int $id
+     * @param bool $show
+     * @throws \coding_exception
+     */
+    public function set_visibility(int $id, bool $show = true) {
+        $instance = $this->get_instance($id);
+        $instance->set('isvisible', $show);
+        $instance->save();
```

Comment:
So here's a bug, if I have an existing content type and I click on the eye icon to hide it, it barfs with
```
Error: Data submitted is invalid

[More information about this error](https://docs.moodle.org/401/en/error/moodle/invalidpersistenterror)
error
Error
Debug info: datasources: Data submitted is invalid
Error code: invalidpersistenterror
error
Error
Stack trace:

    line 558 of /lib/classes/persistent.php: core\invalid_persistent_exception thrown
    line 607 of /lib/classes/persistent.php: call to core\persistent->update()
    line 371 of /mod/cms/classes/manage_content_types.php: call to core\persistent->save()
    line 83 of /mod/cms/classes/manage_content_types.php: call to mod_cms\manage_content_types->set_visibility()
    line 36 of /mod/cms/managetypes.php: call to mod_cms\manage_content_types->execute()
```
However, if I edit the content type and save it from there, then I can click on the hide/show icon, and then it works